app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+
+    if user &.authenticate(params[:session][:password])
```

Comment:
toản tử save navigation `&.` có tác dụng gì em?
```suggestion
    if user&.authenticate(params[:session][:password])
```

Reply:
Toán tử save navigation &. được sử dụng để kiểm tra xem object user có nill hay không, nó được xuất bản lần đầu tiên trong phiên bản ruby 2.3.0.
Ngoài ra trong Rails còn cung cấp method try() để kiểm tra object có nil không. 
- &. chỉ hoat động khi object nil còn trường hợp false sẽ raise exception. Trong khi try() sẽ trả về nil ngay cả khi object là false.
- Đối với try() thì nó sẽ trả về nil khi truy cập đến method không tồn tại. Còn .& thì sẽ raise exception.
- try nên được dùng khi không đảm bảo method được gọi có tồn tại hay không.
- &. nên được dùng khi không đảm bảo object có bị nil hay không.